discovery/azure/azure.go

```diff
@@ -564,6 +574,10 @@ func mapFromVMScaleSetVM(vm compute.VirtualMachineScaleSetVM, scaleSetName strin
 		computerName = *(vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName)
 	}
 
+	if vm.VirtualMachineScaleSetVMProperties != nil && vm.VirtualMachineScaleSetVMProperties.HardwareProfile != nil {
+		size = string(vm.VirtualMachineScaleSetVMProperties.HardwareProfile.VMSize)
+	}
```

Comment:
Same thing here. And while you're at it could you add the `vm.VirtualMachineProperties.OsProfile.ComputerName != nil` condition to the first if statement, like it is in the other place (so it would be `if vm.VirtualMachineProperties.OsProfile != nil && vm.VirtualMachineProperties.OsProfile.ComputerName != nil`). Not really related to this PR but should done.